trlx/trainer/accelerate_ppo_trainer.py

```diff
@@ -452,7 +472,17 @@ def make_experience(self, num_rollouts: int = 1024, iter_count: int = 0):  # noq
 
             for sample_idx in range(n_samples):
                 rewards = kl_penalty[sample_idx]
-                rewards[-1] += scores[sample_idx].cpu()
+                # Then add in rewards
+                if scores.shape[1] == 1:
+                    # NOTE: Final reward given at EOS token following HHH practice
+                    rewards[-1] += scores[sample_idx][0].cpu()
+                else:
+                    score = scores[sample_idx]
+                    score_right_padding = torch.sum(scores_mask[sample_idx])
+                    score = score[:score_right_padding].cpu()
+                    p_score = torch.zeros_like(rewards)
+                    p_score[: score.shape[0]] += score
+                    rewards += p_score
```

Comment:
Nit: could you drop a comment here for what a `p_score` is, and why these computations are needed

Reply:
This is to guard against sizing errors due to the tokenisation issues we discussed today. It is fixed in a later pr.